filters/auth/authclient.go

```diff
@@ -92,9 +95,39 @@ func (ac *authClient) doClonedGet(ctx filters.FilterContext) (int, error) {
 	return rsp.StatusCode, nil
 }
 
+type authClientDialerDialer struct {
+	net.Dialer
+	f func(ctx context.Context, network, addr string) (net.Conn, error)
+}
+
+func newAuthClientDialer(d net.Dialer) *authClientDialerDialer {
+	return &authClientDialerDialer{
+		Dialer: d,
+		f:      d.DialContext,
+	}
+}
+
+// DialContext wraps net.Dialer's DialContext to add logs to the
+// current opentracing span and returns an error, if there is an error
+// while dialing.
+func (acdd *authClientDialerDialer) DialContext(ctx context.Context, network, addr string) (net.Conn, error) {
+	span := opentracing.SpanFromContext(ctx)
+	if span != nil {
```

Comment:
span is always nil as far as I tested :(

Reply:
the problem was caused by not having:

	req = req.WithContext(opentracing.ContextWithSpan(req.Context(), parentSpan))